src/main/java/uk/gov/hmcts/divorce/noticeofchange/event/SystemApplyNoticeOfChange.java

```diff
@@ -120,17 +124,35 @@ private void updateChangeOfRepresentation(CaseDetails<CaseData, State> details,
 
         if (APPLICANT_1_SOLICITOR.getRole().equals(changeOrganisationRequest.getCaseRoleId().getRole())) {
             updateOrgPolicyAndSolicitorDetails(applicant1Solicitor,
-                    nocRequestingUser, loggedInUserEmail);
+                    nocRequestingUser, nocSolicitorOrgName, loggedInUserEmail);
+            setApplicantRepresented(details.getData().getApplicant1());
         } else {
             updateOrgPolicyAndSolicitorDetails(applicant2Solicitor,
-                    nocRequestingUser, loggedInUserEmail);
+                    nocRequestingUser, nocSolicitorOrgName, loggedInUserEmail);
+            setApplicantRepresented(details.getData().getApplicant2());
         }
     }
 
-    private static void updateOrgPolicyAndSolicitorDetails(Solicitor applicantSolicitor,
-                                                           ProfessionalUser nocRequestingUser, String loggedInUserEmail) {
-
+    private static void updateOrgPolicyAndSolicitorDetails(
+        Solicitor applicantSolicitor, ProfessionalUser nocRequestingUser,
+        String nocRequestingUserOrgName, String loggedInUserEmail
+    ) {
         applicantSolicitor.setName(String.join(" ", nocRequestingUser.getFirstName(), nocRequestingUser.getLastName()));
         applicantSolicitor.setEmail(loggedInUserEmail);
+        applicantSolicitor.setFirmName(nocRequestingUserOrgName);
+    }
+
+    private static void setApplicantRepresented(Applicant applicant) {
+        applicant.setSolicitorRepresented(YES);
+        applicant.setOffline(NO);
+    }
+
+    public static void resetConditionalOrderFields(CaseData data) {
+        data.getConditionalOrder().getConditionalOrderApplicant1Questions().setIsSubmitted(NO);
+        data.getConditionalOrder().getConditionalOrderApplicant1Questions().setIsDrafted(NO);
+        if (!data.getApplicationType().isSole()) {
+            data.getConditionalOrder().getConditionalOrderApplicant2Questions().setIsSubmitted(NO);
+            data.getConditionalOrder().getConditionalOrderApplicant2Questions().setIsDrafted(NO);
+        }
```

Comment:
I don't understand why they are doing this though, is it a hack?

Reply:
There are some conditions based on these flags, for example, event show conditions:
[DraftConditionalOrder](https://github.com/hmcts/nfdiv-case-api/blob/2ce9b759ffdf8fae487da34d8e291b658866d8de/src/main/java/uk/gov/hmcts/divorce/common/event/DraftConditionalOrder.java#L97)
[SubmitConditionalOrder](https://github.com/hmcts/nfdiv-case-api/blob/2ce9b759ffdf8fae487da34d8e291b658866d8de/src/main/java/uk/gov/hmcts/divorce/common/event/SubmitConditionalOrder.java)

And we set them early in the process when sols start an application:
https://github.com/hmcts/nfdiv-case-api/blob/2ce9b759ffdf8fae487da34d8e291b658866d8de/src/main/java/uk/gov/hmcts/divorce/solicitor/event/page/SolHowDoYouWantToApplyForDivorce.java#L81-L86

I think they had issues when citizen -> sol was done because these flags weren't set with defaults appropriately and the sol couldn't progress the case. In the sol -> sol scenario, it also allows the new sol who takes over a case before CO submission to draft a new CO if they want to - case [1657200957061229](https://manage-case.platform.hmcts.net/cases/case-details/1657200957061229#History) is a recent example of this in prod. I've pointed this redrafting out to Kim and she seems happy with it.

We could look to simplify the conditions in some way for the new NoC, but initially making it consistent with what we've had in prod for roughly 20 months with caseworker NoC seems lower risk.

Reply:
ah, that makes sense to me now. I guess if it's on the caseworker noc with no explanation it's just matching so can see why you didn't comment